base/array.jl

```diff
@@ -1805,7 +1805,7 @@ julia> filter(isodd, a)
 """
 filter(f, As::AbstractArray) = As[map(f, As)::AbstractArray{Bool}]
 
-function filter!(f, a::Vector)
+function filter!(f, a::AbstractVector)
     insrt = 1
```

Comment:
This will only work for 1-based linear indexing arrays. Would be better to iterate over `eachindex(a)`.